gravitee-am-test/specs/management/password-policy.jest.spec.ts

```diff
@@ -0,0 +1,170 @@
+  const adminTokenResponse = await requestAdminAccessToken();
+  accessToken = adminTokenResponse.body.access_token;
+  expect(accessToken).toBeDefined();
+
+  const createdDomain = await createDomain(accessToken, 'domain-idp', faker.company.catchPhraseDescriptor());
+  expect(createdDomain).toBeDefined();
+  expect(createdDomain.id).toBeDefined();
+
+  const domainStarted = await startDomain(createdDomain.id, accessToken);
+  expect(domainStarted).toBeDefined();
+  expect(domainStarted.id).toEqual(createdDomain.id);
+
+  domain = domainStarted;
+});
+
+describe('password policy management', () => {
+  it('must return no password policy', async () => {
+    const passwordPolicies = await getAllPasswordPoliciesRaw(domain.id, accessToken);
+    expect(passwordPolicies.raw.status).toEqual(204);
+  });
+
+  it('must create password policy', async () => {
+    const passwordPolicy = await createPasswordPolicy(domain.id, accessToken, {
+      name: 'default',
+    });
+    expect(passwordPolicy).toBeDefined();
+    expect(passwordPolicy.id).toEqual(passwordPolicy.id);
+    expect(passwordPolicy.defaultPolicy).toBeTruthy();
+  });
+
+  it('must update password policy', async () => {
+    const oldPasswordPolicy = await createPasswordPolicy(domain.id, accessToken, {
+      name: 'new',
+    });
+    const updatedPasswordPolicy = await updatePasswordPolicy(domain.id, accessToken, oldPasswordPolicy.id, {
+      name: 'updated-new',
+      defaultPolicy: true,
+    });
+    expect(updatedPasswordPolicy).toBeDefined();
+    expect(updatedPasswordPolicy.id).toEqual(updatedPasswordPolicy.id);
+    expect(updatedPasswordPolicy.defaultPolicy).toBeTruthy();
+  });
+
+  it('must find all password policies', async () => {
+    const idpSet = await getAllPasswordPolicies(domain.id, accessToken);
+    expect(idpSet.length).toBeGreaterThan(0);
+  });
+
+  it('must delete password policy', async () => {
+    const passwordPolicy = await createPasswordPolicy(domain.id, accessToken, {
+      name: 'policy-to-delete',
+    });
+    await deletePasswordPolicy(domain.id, accessToken, passwordPolicy.id);
+    await expect(async () => {
+      await getPasswordPolicyRaw(domain.id, accessToken, passwordPolicy.id);
+    }).rejects.toThrow();
+  });
+
+  it('must not delete password policy when not found', async () => {
+    await expect(async () => {
+      await deletePasswordPolicy('not-found-id', accessToken, 'not-found-id');
+    }).rejects.toThrow();
+  });
+
+  it('must assign password policy to identity provider', async () => {
+    const policy = await createPasswordPolicy(domain.id, accessToken, {
+      name: 'default-to-assign',
+    });
+    await createIdp(domain.id, accessToken, getIdp());
+    await setPasswordPolicyDefault(domain.id, accessToken, policy.id);
+
+    const updatedPolicy = await getPasswordPolicy(domain.id, accessToken, policy.id);
+    expect(updatedPolicy).toBeDefined();
+    expect(updatedPolicy.defaultPolicy).toBeTruthy();
+  });
```

Comment:
**todo**: either the test description is not good or the test is not good. Description says we are assigning policy to an IDP but after creating the IDP, we only create a default policy, I can't see the request to assign the policy to the IDP

Reply:
the description was incorrect. Fixed.